test/Microsoft.ML.Functional.Tests/ModelFiles.cs

```diff
@@ -30,6 +32,80 @@ private class InputData
             public float[] Features { get; set; }
         }
 
+        /// <summary>
+        /// Model Files: The (minimum) nuget version can be found in the model file.
+        /// </summary>
+        [Fact]
+        public void DetermineNugetVersionFromModel()
+        {
+            var modelFile = GetDataPath($"backcompat{Path.DirectorySeparatorChar}keep-model.zip");
+            var versionFileName = @"TrainingInfo\Version.txt"; // Must use '\' for cross-platform testing.
+            using (ZipArchive archive = ZipFile.OpenRead(modelFile))
+            {
+                // The version of the entire model is kept in the version file.
+                var versionPath = archive.Entries.First(x => x.FullName == versionFileName);
+                Assert.NotNull(versionPath);
+                using (var stream = versionPath.Open())
+                using (var reader = new StreamReader(stream))
+                {
+                    // The only line in the file is the version of the model.
+                    var line = reader.ReadLine();
+                    Assert.Equal(@"1.0.0.0", line);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Model Files: Save a model, including all transforms, then load and make predictions.
+        /// </summary>
+        /// <remarks>
+        /// Serves two scenarios:
+        ///  1. I can train a model and save it to a file, including transforms.
+        ///  2. Training and prediction happen in different processes (or even different machines). 
+        ///     The actual test will not run in different processes, but will simulate the idea that the 
+        ///     "communication pipe" is just a serialized model of some form.
+        /// </remarks>
+        [Fact]
+        public void FitPipelineSaveModelAndPredict()
+        {
+            var mlContext = new MLContext(seed: 1);
+
+            // Get the dataset.
+            var data = mlContext.Data.LoadFromTextFile<HousingRegression>(GetDataPath(TestDatasets.housing.trainFilename), hasHeader: true);
+
+            // Create a pipeline to train on the housing data.
+            var pipeline = mlContext.Transforms.Concatenate("Features", HousingRegression.Features)
+                .Append(mlContext.Regression.Trainers.FastTree(
+                    new FastTreeRegressionTrainer.Options { NumberOfThreads = 1, NumberOfTrees = 10 }));
+
+            // Fit the pipeline.
+            var model = pipeline.Fit(data);
+
+            var modelPath = DeleteOutputPath("fitPipelineSaveModelAndPredict.zip");
+            // Save model to a file.
+            using (var file = File.Create(modelPath))
```

Comment:
>Create [](start = 35, length = 6)

There are overloads that take the file path, would this be an appropriate place to use these?